### DISCUSSION TITLE
[Feature Request] Automatically set what being selected as the alias of the link when pasting the link.

### DISCUSSION THREAD

**Discussion by xmbai9507** [jglev/obsidian-carry-forward#25] [Ideas]
The link text (maybe call it "link alias" is more clear to users) comes from config or clipboard or selection in the source, and I expect a fourth option, which is from what I select not in the source but in the current note.
For example, if having copied a link to a particular block, in the current note I'd like to type something as the link alias, let's say, "test name", and it would be handy to paste the link in the final form of {{LINK|test name}} just by selecting the "test name".
This feature can be merged into the commands like `Copy link to line (default link text)`. If there is no selection in the current note, use the default hard-coded link text, otherwise, just use what is selected as the link text. In other words, this feature can be regarded as an easier way to quickly set a link alias or maybe even change the existing link alias.
Also thanks for building this awesome plugin, it is just what I'm looking for.

**Comment by jglev** [comment]
Thank you for writing!

I see what you mean, and agree that this could be a useful addition to the plugin. This idea seems conceptually similar to the [Paste URL into Selection](https://github.com/denolehov/obsidian-url-into-selection) plugin -- does that seem accurate for what you're describing?

I do want to think about this a bit more before implementing it, specifically because Carry Forward supports using multiple cursors. E.g., If a user uses multiple cursors and pastes, and there are multiple URLs in the clipboard, how should the plugin behave? Or if a user uses multiple cursors and there are a different number of URLs in the clipboard? Etc.

Then again, it may be enough just to support a single URL and a single cursor to start. If that's the case, I'm wondering whether the Paste URL into Selection plugin already provides relevant functionality.

So, that's all to say, I'm thinking through this more, but it is on my mind!

**Reply by xmbai9507** [reply]
Thanks for your response. 
Maybe a separator character could be introduced. For example, when `alias1; alias2; alias3` is selected, after executing provided command, they will be replaced by `[[LINK1|alias1]]; [[LINK2|alias2]]; [[LINK3|alias3]]` when there is the same number of URLs in the clipboard.
